src/types/mapping.ts

```diff
@@ -0,0 +1,20 @@
+import { DBClient } from '../db/db';
+
+import { ArtistProfile } from './artist';
+import { NFT, NftFactory } from './nft'
+import { ProcessedTrack } from './track';
+
+export type TrackMapping = {
+  [trackId: string]: NFT[]
+}
+
+export type NFTtoTrackIdSource = {
+  nfts: NFT[],
+  dbClient?: DBClient,
+  apiTracksByNFT?: any,
+  contract?: NftFactory
+}
+
+export type MapNFTsToTrackIds = (nftToTrackIdSource: NFTtoTrackIdSource) => TrackMapping;
```

Comment:
NFTtoTrackIdSource and nftToTrackIdSource - "source" seems like too specific a name here, kinda confusing i think? maybe just something simple like "input" is better, so maybe NFTtoTrackIdsInput and nftToTrackIdInput?

Also: For the actual variable name i'd just simplify and call it "input" since in the context it's being used, it's clear that this is the input to the map function. (golang has pretty nice conventions for this https://murfitt.net/blog/short-variable-names-in-go)

Also: The dbClient should be removed now that is is no longer a Promise, since it shouldn't be possible to call the db. I think you'll be fine just removing this without breaking anything, since I don't think it's being used anymore?

```suggestion
export type NFTtoTrackIdsInput = {
  nfts: NFT[],
  apiTracksByNFT?: any,
  contract?: NftFactory
}

export type MapNFTsToTrackIds = (input: NFTtoTrackIdsInput) => TrackMapping;
```

Reply:
Great catch for nuking the db in this object. And I like the naming change, much cleaner.

Awesome on all fronts! 💯